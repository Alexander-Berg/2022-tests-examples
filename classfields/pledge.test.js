const preparer = require('./pledge');

const testForStatus = ({ isCardPage, result, status }) => {
    it(`status === ${ status }`, () => {
        const item = { status };
        const data = preparer({ item, result, isCardPage });

        expect(data).toMatchSnapshot();
    });
};

describe('карточка', () => {
    describe('pledge возвращает нормальные тексты', () => {
        [ '""', 'OK', 'ERROR', 'UNKNOWN' ].forEach((status) => {
            testForStatus({ status, isCardPage: true });
        });
    });
});

describe('полный отчет', () => {
    describe('pledge возвращает нормальные тексты', () => {
        [ '""', 'OK', 'ERROR', 'UNKNOWN' ].forEach((status) => {
            testForStatus({ status, isCardPage: false });
        });
    });
});
