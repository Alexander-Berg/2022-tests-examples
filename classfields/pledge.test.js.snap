// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`карточка pledge возвращает нормальные тексты status === "" 1`] = `
Object {
  "key": "Поиск данных о залоге",
}
`;

exports[`карточка pledge возвращает нормальные тексты status === ERROR 1`] = `
Object {
  "key": "Есть данные о залоге",
}
`;

exports[`карточка pledge возвращает нормальные тексты status === OK 1`] = `
Object {
  "key": "Данные о залоге не найдены",
}
`;

exports[`карточка pledge возвращает нормальные тексты status === UNKNOWN 1`] = `
Object {
  "key": "Данные о залоге появятся чуть позже",
}
`;

exports[`полный отчет pledge возвращает нормальные тексты status === "" 1`] = `undefined`;

exports[`полный отчет pledge возвращает нормальные тексты status === ERROR 1`] = `
Object {
  "key": "Данные о залоге",
  "value": "Есть записи",
}
`;

exports[`полный отчет pledge возвращает нормальные тексты status === OK 1`] = `
Object {
  "key": "Данные о залоге",
  "value": "Нет записей",
}
`;

exports[`полный отчет pledge возвращает нормальные тексты status === UNKNOWN 1`] = `
Object {
  "key": "Данные о залоге",
  "value": "Информация появится чуть позже",
}
`;
